### PR TITLE
CON-3613 - Migrate 'Confirm apprentice deletion' to Approvals

### PR DESCRIPTION
**Pull request by VasanthaKasirajan3008**

### CODE REVIEW

File: src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Controllers/ApprenticeController.cs [hunk 1]
```diff
@@ -249,5 +254,38 @@ public IActionResult Sent()
             var model = TempData[nameof(ConfirmViewModel.NewEmployerName)] as string;
             return View(nameof(Sent), model);
         }
+
+        [HttpGet]
+        [Route("{CohortReference}/{apprenticeshipHashedId}/Delete", Name = RouteNames.ApprenticeDelete)]
```
Comment (cofaulco) [inline src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Controllers/ApprenticeController.cs:259]: adding cohorReference to the route here doesn't really make sense and is inconsistent

File: src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Controllers/ApprenticeController.cs [hunk 2]
```diff
@@ -249,5 +254,38 @@ public IActionResult Sent()
...
+        [Route("{CohortReference}/{apprenticeshipHashedId}/Delete", Name = RouteNames.ApprenticeDelete)]        
+        [Authorize(Policy = nameof(PolicyNames.HasAccountOwnerPermission))]
+        public async Task<ActionResult> DeleteConfirmation(DeleteConfirmationRequest deleteConfirmationRequest)
+        {
+            var viewModel = await _modelMapper.Map<DeleteConfirmationViewModel>(deleteConfirmationRequest);
+            return View(viewModel);
+        }
+
+        [HttpPost]        
+        [Route("{CohortReference}/{apprenticeshipHashedId}/Delete", Name = RouteNames.ApprenticeDelete)]
```
Comment (cofaulco) [inline src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Controllers/ApprenticeController.cs:268]: and here

File: src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Mappers/Apprentice/DeleteDraftApprenticeshipRequestMapper.cs [hunk 3]
```diff
@@ -0,0 +1,15 @@
+﻿using SFA.DAS.CommitmentsV2.Api.Types.Requests;
+using SFA.DAS.CommitmentsV2.Shared.Interfaces;
+using SFA.DAS.ProviderCommitments.Web.Models.Apprentice;
+using System.Threading.Tasks;
+
+namespace SFA.DAS.ProviderCommitments.Web.Mappers.Apprentice
+{
+    public class DeleteDraftApprenticeshipRequestMapper : IMapper<DeleteConfirmationViewModel, DeleteDraftApprenticeshipRequest>
```
Comment (cofaulco) [inline src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Mappers/Apprentice/DeleteDraftApprenticeshipRequestMapper.cs:8]: is this mapper class necessary. I know it is the convention but in this instance seems a little pointless

File: src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Controllers/ApprenticeController.cs (hunk 1 above)
Comment (cofaulco) [reply src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Controllers/ApprenticeController.cs:259]: this action should be move to DraftApprenticeshipController

Comment (VasanthaKasirajan3008) [reply src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Controllers/ApprenticeController.cs]: moved to DraftApprenticeshipController

File: src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Controllers/ApprenticeController.cs (hunk 2 above)
Comment (VasanthaKasirajan3008) [reply src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Controllers/ApprenticeController.cs]: moved to DraftApprenticeshipController

File: src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Mappers/Apprentice/DeleteDraftApprenticeshipRequestMapper.cs (hunk 3 above)
Comment (VasanthaKasirajan3008) [reply src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Mappers/Apprentice/DeleteDraftApprenticeshipRequestMapper.cs:8]: need the new instance and this way can unit test

Comment (cofaulco) [reply src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Mappers/Apprentice/DeleteDraftApprenticeshipRequestMapper.cs:8]: could just create the new instance in the controller action and unit test would work fine as there is nothing being set so no mocking would be necessary?

**Review by cofaulco — APPROVED** [review]

File: src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Mappers/Apprentice/DeleteDraftApprenticeshipRequestMapper.cs (hunk 3 above)
Comment (VasanthaKasirajan3008) [reply src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Mappers/Apprentice/DeleteDraftApprenticeshipRequestMapper.cs]: could just create the new instance in the controller action and unit test would work fine as there is nothing being set so no mocking would be necessary - done

**Pull request merged**